### ISSUE TITLE
Error  - gym_gazebo2 Usage

### ISSUE THREAD

**Issue by camontblanc** [AcutronicRobotics/gym-gazebo2#26]
**Environment/Algorithm**
I was trying the "python3 gg_random.py -g" example. Actually, whenever I try to build any MARA environment I keep having the same error.

**Describe the bug**
ModuleNotFoundError: No module named 'rclpy._rclpy'.

**Screenshots**
![Error](https://user-images.githubusercontent.com/45245175/59285868-5001e400-8c34-11e9-9cb7-9802f4db8694.png)

**Additional context**
I installed ROS2 Crystal and I have tried almost everything: uninstalling ROS2 and installing it again, same for gym_gazebo2.

**Comment by rkojcev** [comment]
Hi @camontblanc 
You need to source the environments, please use the following [script](https://github.com/AcutronicRobotics/gym-gazebo2/blob/master/provision/mara_setup.sh) to do that.

**Comment by camontblanc** [comment]
Hi @rkojcev !
First, thank you for your answer! Do I have to do that even though I am not using any of the MARA examples?
Basically, what I am doing is running python in the shell, I import gym and gym_gazebo2 (clearly, I have already sourced ROS2 Crystal) but when I do: gym.make('MARA-v0') I have the error I told you before.

Thank you for your time!

**Comment by rkojcev** [comment]
@camontblanc 

calling  `gym.make('MARA-v0')` will want to call the MARA robot and all necessary packages that come with this [repo](https://github.com/AcutronicRobotics/MARA/releases/tag/crystal) 

All the necessary steps are explained in the install from this repo.

**Comment by camontblanc** [comment]
@rkojcev 

Hi! I followed (again) the step in [this repo](https://github.com/AcutronicRobotics/gym-gazebo2/blob/master/INSTALL.md) but I keep having the same error.

I also included in the ~/.bashrc file this [script](https://github.com/AcutronicRobotics/gym-gazebo2/blob/master/provision/mara_setup.sh) but it doesn't work.

**Comment by YueErro** [comment]
Hi @camontblanc,

>Hi! I followed (again) the step in this repo but I keep having the same error.

Make sure you use crystal release in both repos ([gym-gazebo2](https://github.com/AcutronicRobotics/gym-gazebo2/tree/crystal) and [MARA](https://github.com/AcutronicRobotics/MARA/tree/crystal)), master branch is dashing.
I just noticed `wget` are wrong in crystal branch, i would try with this ones instead:
```sh
wget https://raw.githubusercontent.com/AcutronicRobotics/MARA/crystal/mara-ros2.repos
wget https://raw.githubusercontent.com/AcutronicRobotics/gym-gazebo2/crystal/provision/additional-repos.repos
```

**Comment by camontblanc** [comment]
Hi @YueErro ,

I tried these `wget`:

`wget https://raw.githubusercontent.com/AcutronicRobotics/MARA/crystal/mara-ros2.repos
wget https://raw.githubusercontent.com/AcutronicRobotics/gym-gazebo2/crystal/provision/additional-repos.repos`

but when compiling the workspace I get the following:
![Error](https://user-images.githubusercontent.com/45245175/59293411-c575b080-8c44-11e9-9b68-ac5b05429e37.png)

when it is supposed to be:
`35 packages finished [12min 26s]
4 packages had stderr output: cv_bridge orocos_kdl python_orocos_kdl robotiq_gripper_gazebo_plugins`

Eventually, when I run an example:
![ExampleError](https://user-images.githubusercontent.com/45245175/59293537-feae2080-8c44-11e9-86f8-a01c39729c1d.png)

**Comment by camontblanc** [comment]
@rkojcev @YueErro 
I managed to fix the above error. However I did notice these things:
1. I echoed the PYTHONPATH and this is what I got: `/home/carlos/ros2_mara_ws/install/lib/python3.6/site-packages:/home/carlos/ros2_mara_ws/install/lib/python3.7/site-packages:/opt/ros/crystal/lib/python3.6/site-packages:/home/carlos/ros2_mara_ws/install/lib/python3/dist-packages`.
2. However, I'm noticing that in "~/gym-gazebo2/provision/mara_setup.sh" they are doing this: `export PYTHONPATH=$PYTHONPATH:~/ros2_mara_ws/install/lib/python3/dist-packages `
3.  I looked into the folder `~ /ros2_mara_ws/install/lib/`, but there is no `/python3/dist-packages`.

Might this be something?

**Comment by nzlz** [comment]
Hi @camontblanc ,

When you execute `wget https://raw.githubusercontent.com/AcutronicRobotics/MARA/crystal/mara-ros2.repos` you are getting MARA's master `branch`, not `crystal` (check the content of the file). For now, go to _ros2_mara_ws/src/MARA_ and checkout to crystal release (its not a branch, yet). Then,  recompile.

We will fix and test this today and let you know when the update is completed.

**Comment by rkojcev** [comment]
@camontblanc we are doing major refurbish and upgrade to all of our public repos these days in order to comply with different ros2 versions. Once we are done we will let you know and hopefully we will resolve all these inconsistencies trough out our repos.

**Comment by nzlz** [comment]
Created `crystal` branches for every _gym-gazebo2_ related repository. Go to the `crystal` branch of _ros2learn_ or _gym-gazebo2_, depending what you want to install, and follow the distribution specific instructions available in the README.

**Comment by rkojcev** [comment]
@camontblanc can you try now and see if everything is ok with the crystal branch of the repositories?

**Comment by camontblanc** [comment]
@nzlz @rkojcev 
First of all, thank you very much for your help. Right now, I'm following these instructions: [crystal branch](https://github.com/AcutronicRobotics/gym-gazebo2/blob/crystal/INSTALL.md). I'll let you know as soon as I'm finished.

**Comment by camontblanc** [comment]
@rkojcev @nzlz 
This is what I get. 
1. After compiling the `~/ros2_mara_ws` workspace and doing the provisioning step, when I run the `/.bashrc` file:
`not found: "/home/carlos/ros2_mara_ws/install/share/orocos_kdl/local_setup.sh"`
`not found: "/home/carlos/ros2_mara_ws/install/share/orocos_kdl/local_setup.bash"`

2. I tried to make `gym.make('MARA-v0)` but I still get the error `ModuleNotFoundError: No module named 'rclpy._rclpy'.`

**Comment by nzlz** [comment]
Regarding n1, you probably forgot to execute the line after the workspace is compiled, double check instructions please. That will get remove these informative errors.

Regarding n2, can you reinstall ros-crystal-desktop? That will give you ros-crystal-rclpy, which you will be able to use when sourcing /opt/ros/crystal/setup.bash via provisioning script.

(sorry for the lack the format, talking from the phone)

**Comment by camontblanc** [comment]
@nzlz 
1. You're right, I forgot `touch ~/ros2_mara_ws/install/share/orocos_kdl/local_setup.sh ~/ros2_mara_ws/install/share/orocos_kdl/local_setup.bash`.
2. I reinstalled ros-crystal-desktop but it still doesn't work. I'm trying to run: `cd ~/gym-gazebo2/examples/MARA` -> `python3 gg_random.py -g` but I get
![Error](https://user-images.githubusercontent.com/45245175/59357530-9e74b880-8cf0-11e9-987c-12ff28b95c11.png)

**Comment by camontblanc** [comment]
@nzlz @rkojcev 

Hi! I deleted the  `~/ros2_mara_ws` and `~/gym_gazebo2` folders from my computer did everything all over again. The workspace compiled successfully and I did the provisioning step. However, the same error still appears (`ModuleNotFoundError: No module named 'rclpy._rclpy'`). As suggested by @nzlz,  I reinstalled ros-crystal-desktop via Debian Packages. I really don't know what else to do given that I've repeated the steps in the INSTALL.md in Crystal Branch many times.

Thanks.
